### PR TITLE
Dragon Egg Update

### PR DESCRIPTION
**Pull request by Leander-0**
## About the PR

First and foremost i want to thank @Stagnation and @GentleButter on the discord server for both inspiration, art and sprites.

I got really annoyed and sad that so many people had to be bwoinked trying to have a space dragon as a pet so here is the solution for said issue, dragon eggs will have a 30% chance of dropping from a dungeon space dragon once butchered.

Said egg dragon will have an automatic ghost role open for it and yes its a **WHITELIST ONLY GHOSTROLE** because no one is having that your terribly hard to keep and maintain baby dragon be inhabited by a dumbass.

It even has an animation for it when it hatches since the way it works it that.

You inhabit the ghost role egg > despawns the original egg transfers your mind and spawns the animation entity > once the animation its over you mind its once again transferred into the new baby dragon entity.

Jank? Unmaintainable? it works, this is the best code i came up with with my lack of experience.

## Why / Balance

New fun pet companion for the salvagers, could be also sold to the highest bidder or just eaten. In the case of the ghost role player having to exit the game and leave the dragon uninhibited then it will act as a mice AI and go and find food when it gets hungry.

**Dragon Stats:**
- Health: 400 to die but gets critical by 250.
- Attack: 5 slash 5 piercing, way less to the 20+ damage from the regular space dragon.
- Fire breath: video shows how damaging it is, at best is going to work as free flares buddy and cigar starter.
- Defense: can survive to space without issues but that is about it, still affected by temperature.

**Egg Stats:**
- Price: 20K
- Food: 12u ichor, 12u nutriment and 12u vitamins.
- Emotional Value: free baby dragon.

You can also even cook it if you feel like cooking an expensive delicacy.
**Recipe:**
- 5u Salt
- 5u Pepper
- 1 whole dragon egg on a bowl vigorously beaten
- 1 whole stick of butter

## How to test

Spawn a bunch of dungeon space dragons kill them and butcher them repeat until an egg spawns and select the role.

## Media

https://github.com/new-frontiers-14/frontier-station-14/assets/80334192/5a021af2-f0e8-4d38-ad23-d7d210a70005

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Coded new "TransferMindOnDespawn" System and Component, meant to be used along the egg spawn animation and made in such a way that you can set the prototype to be spawned as and mind transferred to when the animation ends.

Also ported https://github.com/space-wizards/space-station-14/pull/26746 to give the dragon fire breathing for both the regular dragons and the baby one.

**Changelog**

:cl: Leander
- add: New dragon egg and baby dragon are now more easy to find on this dragon hunting season, don't forget to butcher them!
- add: Dragon DNA evolved and there has been sights of them spitting walls of flames.
- add: New omelette recipe using expensive hard to find ingredients.

### CODE REVIEW

**Comment by dvir001** [comment]
Is this ready now?

**Comment by dvir001** [comment]
The upstream code isnt a small one line or one file PR
Can you split it into a new cherry pick PR by itself please?

**Comment by dvir001** [comment]
Do not merge till
https://github.com/new-frontiers-14/frontier-station-14/pull/1458

Meged

File: Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl [hunk 1]
```diff
@@ -22,3 +22,7 @@ ghost-role-information-ert-mailcarrier-description = Assist with paperworks effo
 
 ghost-role-information-jerma-name = Jerma
 ghost-role-information-jerma-description = Pog moment
+
+ghost-role-information-baby-dragon-name = Space baby dragon
+ghost-role-information-baby-dragon-description = Hatch from your egg and go on incredible adventures with your mom and his crew!
+ghost-role-information-baby-dragon-rules = You are about to hatch from your egg, make sure to know who is your mom, be loyal and protect them!
```
Comment (arimah) [inline Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl:28]: **Nitpick:** `You are about to hatch from your egg. Make sure to know who your mom is, be loyal and protect them!`

Avoids a run-on sentence, and uses correct syntax in the second clause. Also see above re mom's gender: use 'they' or 'she' consistently.

File: Resources/Prototypes/_NF/Misc/dragon_egg.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
+- type: entity
+  parent: FoodEggBase
+  id: DragonEgg
+  name: dragon egg
+  description: How many did you kill just to get one of this eggs.
```
Comment (arimah) [inline Resources/Prototypes/_NF/Misc/dragon_egg.yml:5]: **Nitpick:** this -> these

**Comment by dvir001** [comment]
Please add ``# Frontier`` to any line you added to a yml that isnt under _NF and add ``// Frontier`` to any block of code that isnt under _NF

**Comment by Dysh3s** [comment]
250/400 hp looks a bit too excessive, for a freshly hatched dragon noodle, no?

**Comment by Leander-0** [comment]
> 250/400 hp looks a bit too excessive, for a freshly hatched dragon noodle, no?

Changed to 200/300 still more than a regular human but less than the 350/400 from the dragon and has 100hp to be able to be resuscitated from crit, plus the dragon does not have have a way to get armor or defense from anything so its more or less more exposed to anything.

File: Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl [hunk 3]
```diff
@@ -22,3 +22,7 @@ ghost-role-information-ert-mailcarrier-description = Assist with paperworks effo
 
 ghost-role-information-jerma-name = Jerma
 ghost-role-information-jerma-description = Pog moment
+
+ghost-role-information-baby-dragon-name = Space baby dragon
+ghost-role-information-baby-dragon-description = Hatch from your egg and go on incredible adventures with your mom and his crew!
```
[the remark this answers is not in the document]
Comment (Leander-0) [reply Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl]: resolved on last commit.

File: Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl [hunk 4]
```diff
@@ -22,3 +22,7 @@ ghost-role-information-ert-mailcarrier-description = Assist with paperworks effo
 
 ghost-role-information-jerma-name = Jerma
 ghost-role-information-jerma-description = Pog moment
+
+ghost-role-information-baby-dragon-name = Space baby dragon
```
[the remark this answers is not in the document]
Comment (Leander-0) [reply Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl]: resolved on last commit.

File: Resources/Prototypes/_NF/Entities/Mobs/NPCs/baby_dragon.yml [hunk 5]
```diff
@@ -0,0 +1,172 @@
+- type: entity
+  parent:
+  - SimpleSpaceMobBase
+  - FlyingMobBase
+  id: MobDragonPet
+  name: space baby dragon
```
[the remark this answers is not in the document]
Comment (Leander-0) [reply Resources/Prototypes/_NF/Entities/Mobs/NPCs/baby_dragon.yml]: resolved on last commit.

File: Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl (hunk 1 above)
Comment (Leander-0) [reply Resources/Locale/en-US/_NF/ghost/roles/ghost-role-component.ftl]: resolved on last commit.

File: Resources/Prototypes/_NF/Misc/dragon_egg.yml (hunk 2 above)
Comment (Leander-0) [reply Resources/Prototypes/_NF/Misc/dragon_egg.yml]: resolved on last commit.

File: Resources/Prototypes/Entities/Mobs/Player/dragon.yml [hunk 6]
```diff
@@ -150,6 +150,8 @@
   - type: GuideHelp
     guides:
     - MinorAntagonists
+# - type: GenericAntag #Frontier
+#   rule: Dragon #Frontier
```
Comment (dvir001) [inline Resources/Prototypes/Entities/Mobs/Player/dragon.yml:154]: Remove added lines

File: Resources/Prototypes/Entities/Mobs/Player/dragon.yml [hunk 7]
```diff
@@ -190,11 +192,17 @@
     spawned:
     - id: FoodMeatDragon
       amount: 3
+    - id: DragonEgg
+      amount: 1
+      prob: 0.3
```
Comment (dvir001) [inline Resources/Prototypes/Entities/Mobs/Player/dragon.yml:197]: Missing comment

**Review by dvir001 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/Player/dragon.yml (hunk 6 above)
Comment (Leander-0) [reply Resources/Prototypes/Entities/Mobs/Player/dragon.yml]: solved on last commit

File: Resources/Prototypes/Entities/Mobs/Player/dragon.yml (hunk 7 above)
Comment (Leander-0) [reply Resources/Prototypes/Entities/Mobs/Player/dragon.yml]: solved on last commit

**Comment by dvir001** [comment]
Bless

**Comment by Cheackraze** [comment]
its a bit spaghetti but i see what the issue is

**Review by Cheackraze — APPROVED** [review]

**Pull request merged**